sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/init.sql

```diff
@@ -0,0 +1,312 @@
+        reported_first_session_ping BOOL
+        OPTIONS
+          (description = "True if the client ever reported a first_session ping."),
+          reported_metrics_ping BOOL
+        OPTIONS
+          (description = "True if the client ever reported a metrics ping."),
+          min_first_session_ping_date DATE
+        OPTIONS
+          (description = "Date of the earliest first_session ping reported."),
+          adjust_network__source_ping STRING
+        OPTIONS
+          (description = "Name of the ping that reported the first adjust_network value."),
+          install_source__source_ping STRING
+        OPTIONS
+          (description = "Name of the ping that reports the install_source value."),
+          adjust_network__source_ping_date DATE
+        OPTIONS
+          (description = "Date of the ping that reported the first adjust_network value."),
+          install_source__source_ping_date DATE
+        OPTIONS
+          (description = "Date of the ping that reported the first install_source value.")
+      >
+  )
+PARTITION BY
+  first_seen_date
+CLUSTER BY
+  channel,
+  first_reported_country,
+  adjust_network,
+  install_source
+OPTIONS
+  (
+    description = "First observations for Firefox Android clients retrieved from the earliest pings: baseline, first_session and metrics. The attributes stored in this table include the first attribution, device, OS version and ISP. This table should be accessed through the user-facing view `fenix.firefox_android_clients`. Proposal: https://docs.google.com/document/d/12bj4DhCybelqHVgOVq8KJlzgtbbUw3f68palNrv-gaM/. For more details about attribution and campaign structure see https://help.adjust.com/en/article/tracker-urls#campaign-structure-parameters."
+  );
+
+INSERT
+  `moz-fx-data-shared-prod.fenix_derived.firefox_android_clients_v1`
+WITH first_seen AS (
+  SELECT
+    client_id,
+    first_seen_date
+  FROM
+    `mozdata.org_mozilla_firefox.baseline_clients_first_seen`
+  WHERE
+    submission_date >= '2021-01-01'
+    AND first_seen_date = submission_date
+),
+-- Find earliest first_session ping per client.
+first_session_ping AS (
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(SAFE.PARSE_DATE('%F', SUBSTR(client_info.first_run_date, 1, 10))) AS first_run_date,
+    ANY_VALUE(normalized_country_code) AS first_reported_country,
+    ANY_VALUE(metadata.isp.name) AS first_reported_isp,
+    ANY_VALUE(normalized_channel) AS channel,
+    ANY_VALUE(client_info.device_manufacturer) AS device_manufacturer,
+    ANY_VALUE(client_info.device_model) AS device_model,
+    ANY_VALUE(client_info.os_version) AS os_version,
+    ANY_VALUE(metrics.string.first_session_campaign) AS adjust_campaign,
+    ANY_VALUE(metrics.string.first_session_network) AS adjust_network,
+    ANY_VALUE(metrics.string.first_session_adgroup) AS adjust_ad_group,
+    ANY_VALUE(metrics.string.first_session_creative) AS adjust_creative
+  FROM
+    `mozdata.org_mozilla_firefox.first_session`
```

Comment:
Use `mozdata.fenix.first_session`(based on `org_mozilla_firefox.first_session`) for channels Release, Beta, Nightly. Aurora dataset `mozdata.org_mozilla_fennec_aurora.first_session` is empty for 2021-2022.